### PR TITLE
fix: Update the spec "du_dirs" to filterable

### PR DESCRIPTION
**Pull request by huali027**
Signed-off-by: Huanhuan Li <huali@redhat.com>

### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [x] Is this PR an enhancement?

### Complete Description of Additions/Changes:
fixes: https://github.com/RedHatInsights/insights-core/issues/3383

### CODE REVIEW

**Comment by huali027** [comment]
@xiangce, please help to review, thanks.

File: insights/specs/default.py [hunk 1]
```diff
@@ -214,7 +215,8 @@ def corosync_cmapctl_cmd_list(broker):
     @datasource(HostContext)
     def du_dirs_list(broker):
         """ Provide a list of directorys for the ``du_dirs`` spec to scan """
-        return ['/var/lib/candlepin/activemq-artemis']
+        filters = get_filters(Specs.du_dirs)
```
Comment (xiangce) [inline insights/specs/default.py:218]: @huali027 - would you please move this `datasource` to the `insights/specs/datasource` dir and add a test for it? Thanks.

**Review by xiangce — CHANGES_REQUESTED** [review]

File: insights/specs/default.py (hunk 1 above)
Comment (xiangce) [reply insights/specs/default.py:218]: And adding the `filterable=True` is a kind of `API change` which will break the existing rules depending on it. Generally it requires an approve of "SPEC Change" in the assets repo.

Comment (huali027) [reply insights/specs/default.py:218]: @xiangce, sure, I'll do it.

File: insights/specs/datasources/get_dirs.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+from insights.core.context import HostContext
+from insights.core.plugins import datasource
+from insights.core.filters import get_filters
+from insights.specs import Specs
+
+
+@datasource(HostContext)
+def du_dirs_list(broker):
+    """ Return a list of directories from the spec filter """
+    filters = get_filters(Specs.du_dirs)
```
Comment (xiangce) [inline insights/specs/datasources/get_dirs.py:14]: Better raise a `SkipCompoent` here when the filter list is empty to end the process a bit earlier.

**Review by xiangce — COMMENTED** [review]
And better rename the `get_dirs` to a prop-noun like `dir_list`.

File: insights/specs/datasources/get_dirs.py (hunk 2 above)
Comment (huali027) [reply insights/specs/datasources/get_dirs.py]: updated, please help to review.

**Review by xiangce — APPROVED** [review]

**Pull request merged**